providers/openfeature-go-feature-flag-provider/lib/openfeature/go-feature-flag/error/errors.rb

```diff
@@ -0,0 +1,76 @@
+# Define a custom error class
+require "open_feature/sdk/provider/error_code"
+
+module OpenFeature
+  module GoFeatureFlag
+    class FlagNotFoundError < StandardError
+      attr_reader :response, :error_code, :error_message
+
+      def initialize(response, flag_key)
+        error_message = "Flag not found: #{flag_key}"
+        super(error_message)
+        @response = response
+        @error_code = OpenFeature::SDK::Provider::ErrorCode::FLAG_NOT_FOUND
+        @error_message = error_message
```

Comment:
nitpick: Generally, you want the super call to be the last thing that happens in initialize.

```suggestion
        @response = response
        @error_code = OpenFeature::SDK::Provider::ErrorCode::FLAG_NOT_FOUND
        @error_message = error_message
        super(error_message)
```

Reply:
Thanks for this hint, this is the total opposite as `java` 🤦